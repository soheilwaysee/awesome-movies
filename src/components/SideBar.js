import React, { useCallback } from "react";
import { NavLink } from "react-router-dom";
import styles from "../styles/components/SideBar.module.css";
import routeNames from "../constants/routeNames";
import classNamesJoiner from "../utils/classNamesJoiner";

const SideBar = ({ show, setShowSideBar }) => {
  const onClickHandler = useCallback(
    () => window.innerHeight > window.innerWidth && setShowSideBar(false),
    [setShowSideBar]
  );
  return (
    <div
      onClick={onClickHandler}
      className={classNamesJoiner([styles.wrapper, !show && styles.hide])}
    >
      <NavLink
        data-testid="link"
        className={styles.link}
        activeClassName={styles.active}
        isActive={(_, location) => {
          return [routeNames.homePage, routeNames.movieNowPlaying].some(
            item => item === location.pathname
          );
        }}
        to={routeNames.movieNowPlaying}
      >
        Now Playing
      </NavLink>
      <NavLink
        data-testid="link"
        className={styles.link}
        activeClassName={styles.active}
        to={routeNames.moviePopular}
      >
        Popular
      </NavLink>
      <NavLink
        data-testid="link"
        className={styles.link}
        activeClassName={styles.active}
        to={routeNames.movieTopRated}
      >
        Top Rated
      </NavLink>
      <NavLink
        data-testid="link"
        className={styles.link}
        activeClassName={styles.active}
        to={routeNames.movieUpcoming}
      >
        Upcoming
      </NavLink>

      <NavLink
        className={styles.link}
        data-testid="link"
        activeClassName={styles.active}
        to={routeNames.movieFavorites}
      >
        Favorites
      </NavLink>
      <NavLink
        className={styles.link}
        data-testid="link"
        activeClassName={styles.active}
        to={routeNames.movieWatchlist}
      >
        Watchlist
      </NavLink>
    </div>
  );
};
export default SideBar;
