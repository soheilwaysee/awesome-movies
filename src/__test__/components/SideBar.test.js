import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-dom/extend-expect';
import SideBar from '../../components/SideBar';
import { MemoryRouter } from 'react-router-dom';

afterEach(cleanup);

describe('SideBar Component ', () => {
  it('renders SideBar Component', async () => {
    afterEach(cleanup);
    const SideBarWithRouter = () => (
      <MemoryRouter initialEntries={['/movie/favorites']} initialIndex={0}>
        <SideBar />
      </MemoryRouter>
    );
    const { findAllByTestId, findByText, container } = render(
      <SideBarWithRouter showSideBar setShowSideBar={() => undefined} />
    );
    const links = await findAllByTestId('link');
    expect(links).toHaveLength(6);
    const favorite = await findByText('Favorites');
    expect(favorite).toHaveClass('link active');
    expect(container.firstChild).toMatchSnapshot();
  });
});
