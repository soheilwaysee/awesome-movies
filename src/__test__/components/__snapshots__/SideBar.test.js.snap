// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar Component  renders SideBar Component 1`] = `
<div
  class="wrapper hide"
>
  <a
    class="link"
    data-testid="link"
    href="/movie/now_playing"
  >
    Now Playing
  </a>
  <a
    class="link"
    data-testid="link"
    href="/movie/popular"
  >
    Popular
  </a>
  <a
    class="link"
    data-testid="link"
    href="/movie/top_rated"
  >
    Top Rated
  </a>
  <a
    class="link"
    data-testid="link"
    href="/movie/upcoming"
  >
    Upcoming
  </a>
  <a
    aria-current="page"
    class="link active"
    data-testid="link"
    href="/movie/favorites"
  >
    Favorites
  </a>
  <a
    class="link"
    data-testid="link"
    href="/movie/watchlist"
  >
    Watchlist
  </a>
</div>
`;
